### DISCUSSION TITLE
formset issue with django-select2

### DISCUSSION THREAD

**Discussion by DarnelleF** [codingjoe/django-select2#77] [Q&A] [answered]
hi @codingjoe, i would love your assistance as well. i am also trying to implement select2 with this package django_select2 via formsets
but select2 field is not rendering and nothing is throwing errors. i have installed django_select2, added it to my url root config, installed redis-server and django-redis via pip, and set up the caches in the settings.py. i have init for jquery via apps.js ( this is for the external dependencies section ) 
i must note that {{form.media.js and form.media.css }} load nothing for me and i do not have a context name for the view its all named form

here is a sample of my code
template

```html
<div class="col-md-8">
  <div class="card">


      <div>
      </div>
      <div>

          <form class="card-box" method="POST">
              {% csrf_token %}
              {{ formset.management_form }}
              {% for form in formset %}
              <div class="row form-row spacer">


                  <!-- form variable -->
                  {{ form.as_p }}
                  <div class="input-group-append">
                      <button class="btn btn-success add-form-row">+</button>

                  </div>
              </div>
              {% endfor %}
            <input class="btn btn-primary" type="submit" name="submit" value="Submit">
              <br>

          </form>
       <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
          {{ form.media.js }}




      </div>
  </div>
```

forms.py

```
class ItemS2Widget(s2forms.ModelSelect2Widget):
          search_fields = [
              "names__icontains",
          ]

class RequisitionItemForm(forms.ModelForm):
    class Meta:
        model = RequisitionItem
        fields = ('item', 'requested_quantity', 'requisition')
        widgets = {'requisition': HiddenInput, 'item': ItemS2Widget()}
```

also i am using django 3.1 and python 3.6
any assistance will be greatly appreciated

**Comment by DarnelleF** [comment]
i guess there is no solution to this...........

**Comment by codingjoe** [comment, marked as answer]
Hi @DarnelleF,

Yes, there is an answer, this is actually a very common question. I was simply on vacation and didn't have time to catch up will all my work until now.

If you add new select inputs to the DOM you need to initialize them. We only initialize all select inputs a single time once the DOM is loaded. Everything you inject via JS afterwards, you will need to initialize yourself. Just look into the JS file to see how to initialize new Select2 fields.

Best
Joe

**Reply by darnellefevrier** [reply]
thanks for the reply but i did it with js and removed the django-select2 package. but i still appreciate it.
